pcf8523/pcf8523_test.go

```diff
@@ -0,0 +1,91 @@
+package pcf8523
+
+import (
+	"encoding/hex"
+	"testing"
+	"time"
+	"tinygo.org/x/drivers/tester"
+)
+
+func TestDecToBcd_RoundTrip(t *testing.T) {
+
+	for i := 0; i < 60; i++ {
+		a := bcd2bin(bin2bcd(i))
+		if a != i {
+			t.Logf("not equal: %d != %d", a, i)
+			t.FailNow()
+		}
+	}
+}
+
+func TestDevice_Reset(t *testing.T) {
+	bus := tester.NewI2CBus(t)
+	fake := bus.NewDevice(DefaultAddress)
+
+	dev := New(bus, DefaultAddress)
+
+	err := dev.Reset()
+	assertNoError(t, err)
+
+	assertEquals(t, fake.Registers[rControl1], 0x58)
+}
+
+func TestDevice_SetPowerManagement(t *testing.T) {
+	bus := tester.NewI2CBus(t)
+	fake := bus.NewDevice(DefaultAddress)
+
+	dev := New(bus, DefaultAddress)
+
+	err := dev.SetPowerManagement(PowerManagement_SwitchOver_ModeStandard)
+	assertNoError(t, err)
+
+	assertEquals(t, fake.Registers[rControl3], 0b100<<5)
+}
+
+func TestDevice_SetTime(t *testing.T) {
+	bus := tester.NewI2CBus(t)
+	fake := bus.NewDevice(DefaultAddress)
+
+	dev := New(bus, DefaultAddress)
+
+	pointInTime, _ := time.Parse(time.RFC3339, "2023-09-12T22:35:50Z")
+	err := dev.SetTime(pointInTime)
+	assertNoError(t, err)
+
+	actual := hex.EncodeToString(fake.Registers[rSeconds : rSeconds+7])
+	expected := "50352212020923"
+	assertEquals(t, actual, expected)
+}
+
+func TestDevice_ReadTime(t *testing.T) {
+	bus := tester.NewI2CBus(t)
+	fake := bus.NewDevice(DefaultAddress)
+
+	expectedPointInTime := time.Date(2023, 9, 12, 17, 55, 42, 0, time.UTC)
+	fake.Registers[rSeconds] = 0x42
+	fake.Registers[rMinutes] = 0x55
+	fake.Registers[rHours] = 0x17
+	fake.Registers[rDays] = 0x12
+	fake.Registers[rMonths] = 0x9
+	fake.Registers[rYears] = 0x23
+
+	dev := New(bus, DefaultAddress)
+
+	//when
+	actualPointInTime, err := dev.ReadTime()
+
+	//then
+	assertNoError(t, err)
+	assertEquals(t, actualPointInTime, expectedPointInTime)
+}
+
+func assertNoError(t testing.TB, e error) {
+	if e != nil {
+		t.Fatalf("unexpected error: %v", e)
+	}
+}
+func assertEquals[T comparable](t testing.TB, a, b T) {
+	if a != b {
+		t.Fatalf("%v != %v", a, b)
+	}
+}
```

Comment:
I've tried to use quicktest instead but always run into a `panic: unimplemented: (reflect.Type).IsVariadic()
` error.

Happy to change if somebody can point me towards how to fix that.

Reply:
See https://github.com/tinygo-org/drivers/blob/dev/mcp23017/device_test.go for driver that is using it now.

Reply:
I think you need

```go
c := qt.New(t)
bus := tester.NewI2CBus(c)
```

instead of 

```go
bus := tester.NewI2CBus(t)
```

Reply:
Actually the linked test also doesn't work :face_with_head_bandage: 
```
~> tinygo test -v ./mcp23017
panic: unimplemented: (reflect.Type).IsVariadic()
FAIL    tinygo.org/x/drivers/mcp23017   3.049s
```

I use:
```
tinygo version
tinygo version 0.29.0 linux/amd64 (using go version go1.20.7 and LLVM version 16.0.6)
```

Looking at the sources that is indeed not implemented:
https://github.com/search?q=repo%3Atinygo-org/tinygo%20IsVariadic&type=code

But I see the tests are run against some bleeding edge version where the tests don't fail:
https://github.com/tinygo-org/drivers/actions/runs/6232602259/job/16916095548#step:5:5

But I couldn't find the corresponding tag/branch in the tinygo repo.